CI/wininstaller/installer.iss

```diff
@@ -0,0 +1,733 @@
+      InstallModePage.CheckListBox.Invalidate();
+    end;
+  end;
+ 
+  // Check for Heroes 3 installation paths
+  Heroes3Path := '';
+  Heroes3Path := RegistryQueryPath('SOFTWARE\GOG.com\Games\1207658787', 'path');
+  if Heroes3Path = '' then
+    Heroes3Path := RegistryQueryPath('SOFTWARE\WOW6432Node\GOG.com\Games\1207658787', 'path');
+  if Heroes3Path = '' then
+    Heroes3Path := RegistryQueryPath('SOFTWARE\New World Computing\Heroes of Might and Magic® III\1.0', 'AppPath');
+  if Heroes3Path = '' then
+    Heroes3Path := RegistryQueryPath('SOFTWARE\WOW6432Node\New World Computing\Heroes of Might and Magic® III\1.0', 'AppPath');
+  if Heroes3Path = '' then
+    Heroes3Path := RegistryQueryPath('SOFTWARE\New World Computing\Heroes of Might and Magic III\1.0', 'AppPath');
+  if Heroes3Path = '' then
+    Heroes3Path := RegistryQueryPath('SOFTWARE\WOW6432Node\New World Computing\Heroes of Might and Magic III\1.0', 'AppPath');
+    
+  // Paths in VCMI directory
+  VCMIFolder := GlobalUserDocsFolder + '\' + '{#VCMIFilesFolder}';
+  MapsFolder := VCMIFolder + '\Maps';
+  DataFolder := VCMIFolder + '\Data';
+  Mp3Folder := VCMIFolder + '\Mp3';
+
+  // Create a custom label for the footer message
+  FooterLabel := TLabel.Create(WizardForm);
+  FooterLabel.Parent := WizardForm;
+  FooterLabel.Caption := 'VCMI v' + '{#AppVersion}' + '.' + '{#AppBuild}';
+  // Padding from the left edge
+  FooterLabel.Left := 10;
+  // Adjust to leave space for multiple lines
+  FooterLabel.Top := WizardForm.ClientHeight - 30;
+  // Adjust for padding
+  FooterLabel.Width := WizardForm.ClientWidth - 20;
+  // Adjust height to accommodate multiple lines
+  FooterLabel.Height := 40; 
+end;
+
+
+function ShouldSkipPage(PageID: Integer): Boolean;
+begin
+  Result := False; // Default is not to skip the page
+
+  if IsUpgrade then
+  begin
+    if (PageID = wpLicense) or (PageID = wpSelectTasks) or (PageID = wpReady) then
+    begin
+      Result := True; // Skip these pages during upgrade
+      Exit;
+    end;
+  end;
+end;
+
+
+procedure CurPageChanged(CurPageID: Integer);
+begin
+  // Ensure the footer message is visible on every page
+  FooterLabel.Visible := True;
+    
+end;
+
+
+function NextButtonClick(CurPageID: Integer): Boolean;
+begin
+  // Skip the custom page on upgrade
+  if IsUpgrade and Assigned(InstallModePage) and (CurPageID = InstallModePage.ID) then
+  begin
+    Result := True;
+    Exit;
+  end;
+
+  // Handle logic for the custom page if it exists
+  if Assigned(InstallModePage) and (CurPageID = InstallModePage.ID) then
+  begin
+    if (InstallModePage.SelectedValueIndex = 0) and not IsAdmin then
+    begin
+      Result := False;
+      Exit;
+    end;
+
+    if InstallModePage.SelectedValueIndex = 0 then
+      WizardForm.DirEdit.Text := GetCommonProgramFilesDir + '\VCMI'
+    else
+      WizardForm.DirEdit.Text := GlobalUserAppdataFolder + '\VCMI';
+  end;
+
+  Result := True;
+end;
+
+
+procedure PerformHeroes3FileCopy();
```

Comment:
we already have this logic in the launcher, not sure it should be copied to installer

Reply:
I have added it to simplify setup process. Not sure why this behavior is 2-3 times quicker on testing machine instead of copying using launcher.

Reply:
@IvanSavenko @Laserlicht what do you think?

Reply:
I have added it according to Discrord discussions few days back. Also it matches logic uninstall wizard.
And is useful only for players with already installed H3, if not installed user still must do this in launcher.

Reply:
Not against it I guess? 
Not sure how it is done currently, but personally I would have added it as selected by default checkbox, alongside with other options - if only to indicate what exactly installer is doing.

Reply:
It's invisible feature. If installed path is present in registry and files not already exist in My games\VCMI, then they are automatically copied to proper place. It's basic check if folders Data, Maps, Mp3 folder is bigger than 1 MB. That copy process is done during "Extracting files" - at the begining of installation. It's too quick in my tests so I can't screenshot it.
But it's done on this screen at the beginning.
![image](https://github.com/user-attachments/assets/213a45c4-d4a8-4143-b115-507b0ccef6f6)

I have tested this info screen notification about file copy.
![image](https://github.com/user-attachments/assets/96707029-d9e0-4dab-9cb6-a93929df238a)


Adding another checkbox section when H3 installation is possible. But it will require installer resizing, bacause it becomes invisible and user need to scroll.
![image](https://github.com/user-attachments/assets/9d7a856e-8c68-4554-ae99-31d03c23c283)

![image](https://github.com/user-attachments/assets/8a268ad3-a98a-45ce-859a-efabcb052b54)

Also don't know what could be reason to uncheck that checkbox. 

For me looks more clear just to inform user that it will be done automatically in "Ready to Install" page

Reply:
What about reducing number of groups?
System integration:
- Associate h3m
- Associate vmap 
- Created desktop shortcuts
- Create start menu shortcuts
VCMI configuration
- Add firewall rules
- Copy H3 data

Reply:
Great idea! Checkbox will appear only if H3 installation is detected
![image](https://github.com/user-attachments/assets/1ee306e1-91f3-4bb2-abc6-7123e41ed9b5)

I have also added warning to unchecking firewall rule
![image](https://github.com/user-attachments/assets/cb57cb24-94da-4ca1-a604-2a48f0239d57)

Will redo all languages strings and add all modded files to PR